clients/drcachesim/tracer/tracer.cpp

```diff
@@ -2902,6 +2974,26 @@ event_thread_exit(void *drcontext)
     if (BUF_PTR(data->seg_base) != NULL) {
         /* This thread was *not* filtered out. */
 
+#ifdef BUILD_PT_TRACER
+        if (op_offline.get_value() && op_enable_kernel_tracing.get_value()) {
+            int recording_sysnum = data->syscall_pt_trace.get_recording_sysnum();
+            if (recording_sysnum != SYS_exit && recording_sysnum != SYS_exit_group) {
+                NOTIFY(0, "ERROR: The last recorded syscall of thread T%d is %d.\n",
+                       dr_get_thread_id(drcontext), recording_sysnum);
+                ASSERT(recording_sysnum == SYS_exit || recording_sysnum == SYS_exit_group,
+                       "recording syscall is not exit");
+            }
+
+            if (!data->syscall_pt_trace.stop_syscall_pt_trace()) {
```

Comment:
We cannot really trace SYS_exit right? We should just skip it in the pre_syscall event.

Reply:
Done.